File: 2020/day1/main.py
#!/usr/bin/python3
import sys

def puzzle1(c):
    for i1 in c:
        for i2 in c:
            if i1+i2 == 2020:
                print(f"{i1} * {i2} = {i1*i2} ")

def puzzle2(c):
    for i1 in c:
        for i2 in c:
            for i3 in c:
                if i3+i1+i2 == 2020:
                    print(f"{i1} * {i2} * {i3} = {i3*i1*i2} ")



def main():
    name = sys.argv[1]
    print(f"reading file {name}")
    
    content = []
    for it in open(name):
        content.append(int(it))

    puzzle1(content)
    puzzle2(content)


if __name__ == "__main__":
    main()


# ________________________________________________________
# Executed in   51,64 secs   fish           external
   # usr time   50,69 secs    2,02 millis   50,68 secs
   # sys time    0,49 secs   77,37 millis    0,42 secs

File: 2020/day1/src/main.rs
use std::env;
use std::fs;

fn puzzle2(numbers: &Vec<i32>) {
    for number1 in numbers.iter() {
        for number2 in numbers.iter() {
            for number3 in numbers.iter() {
                if number1 + number2 + number3 == 2020 {
                    println!(
                        "{} * {} * {} = {}",
                        number1,
                        number2,
                        number3,
                        number3 * number1 * number2
                    )
                }
            }
        }
    }
}

fn puzzle1(numbers: &Vec<i32>) {
    for number1 in numbers.iter() {
        for number2 in numbers.iter() {
            if number1 + number2 == 2020 {
                println!("{} * {} = {}", number1, number2, *number1 * number2)
            }
        }
    }
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let filename = &args[1];
    println!("reading file {}", filename);

    let contents = fs::read_to_string(filename).expect("Something went wrong reading the file");

    let results: Vec<i32> = contents.lines().map(|x| x.parse().unwrap()).collect();
    //println!("{:?}", results);

    puzzle1(&results);
    puzzle2(&results);
}

// ________________________________________________________
//      Executed in    8,08 secs   fish           external 
//         usr time    7,38 secs   53,27 millis    7,33 secs 
//         sys time    0,69 secs   38,52 millis    0,65 secs 


